tools/ci_connector_ops/ci_connector_ops/pipelines/contexts.py

```diff
@@ -7,26 +7,113 @@
 from enum import Enum
 from types import TracebackType
 from typing import Optional
+from abc import ABC
 
 from anyio import Path
 from asyncer import asyncify
+from dagger import Client, Directory
+
 from ci_connector_ops.pipelines.actions import remote_storage, secrets
 from ci_connector_ops.pipelines.bases import ConnectorTestReport
 from ci_connector_ops.pipelines.github import update_commit_status_check
 from ci_connector_ops.pipelines.utils import AIRBYTE_REPO_URL
 from ci_connector_ops.utils import Connector
-from dagger import Client, Directory
 
 
 class ContextState(Enum):
-    INITIALIZED = {"github_state": "pending", "description": "Tests are being initialized..."}
-    RUNNING = {"github_state": "pending", "description": "Tests are running..."}
-    ERROR = {"github_state": "error", "description": "Something went wrong while running the tests."}
-    SUCCESSFUL = {"github_state": "success", "description": "All tests ran successfully."}
-    FAILURE = {"github_state": "failure", "description": "Test failed."}
+    INITIALIZED = {"github_state": "pending", "description": "Pipelines are being initialized..."}
+    RUNNING = {"github_state": "pending", "description": "Pipelines are running..."}
+    ERROR = {"github_state": "error", "description": "Something went wrong while running the Pipelines."}
+    SUCCESSFUL = {"github_state": "success", "description": "All Pipelines ran successfully."}
+    FAILURE = {"github_state": "failure", "description": "Pipeline failed."}
+
+
+class PipelineContext(ABC):
```

Comment:
This is an abstract class but you created instance of it in `metadata_service`. 

I suggest you create a `MetadataPipelineContext(PipelineContext)`. I would also declare `__aenter__` and `__aexit__` as abstract methods, and you make your   `MetadataPipelineContext` implement these methods.
I we eventually spot a pattern on a third child class we could consider unabstracting these methods and implement a sensible default one.

Reply:
Hmm good callout.

I want to push back slightly. But only because I may not fully understand.

But would sensible defaults be what we have
1. On error -> set commit status to error
2. On success -> set commit status to success

For the pipelines im intending to implement I think thats all I need?

These are the tests im thinking about
1. Run metadata lib unit test -> pass/fail
2. Run metadata orchestrator unit test -> pass/fail 
3. Run metadata validator -> pass/fail